nav2_util/include/nav2_util/occupancy_grid.hpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2020 Samsung Research Russia
```

Comment:
Delete these files. We're not going to duplicate all of this code from costmap_2d. We already have a `nav2_utils::costmap` this is completely redundant. I don't see why any of this is necessary.

Reply:
Maybe I am wrong, but I did not find any similar with Costmap2D API for OccupancyGrid maps. So each time when we want to calculate index of OccupancyGrid, or conver mapToWorld/worldToMap coordinates for it, it is required to write/repeat own logic similar to Costmap2D has. Why do not make the same as for Costmap2D, but for OccupancyGrid API in `nav2_util`?

Reply:
I don't want to embed any more 2D information in the stack outside of navigation2 and the recovery server using the nav2_util::Costmap type. Long term we're moving away from 2D maps so I do not want to extend the tentacles of 2D occupancy grids further into the stack that will make it increasingly harder to remove.

I don't understand. Everything in Costmap2D has access to those conversion functions, there's no need to add more. 

My understanding is that this started as a result of wanting to have the constants defined in a common place. You're more than welcome to make nav2_utils have an occupancy grid threshold constants file, but that's the extent of it ideally. Arguably, they're already here in https://github.com/ros-planning/navigation2/blob/master/nav2_util/src/costmap.cpp#L26-L31 just needs to add unknown, though that location is meant for tests. I think you could make a thresholds file, change this file to use that one (and add medium, which shouldn't exist anywhere else) and then use that in all the other places.